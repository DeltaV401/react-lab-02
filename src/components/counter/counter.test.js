'use strict';

import React from 'react';
import renderer from 'react-test-renderer';
import Counter from './counter';

describe('counter tests', () => {
  it('can initially render', () =>{
    // Arrange
    let app = shallow(<Counter />);
    
    // Assert
    expect(app.find('section.counter').exists()).toBe(true);
  });

  it('has a counter that starts at 0', () => {
    // Arrange
    let app = shallow(<Counter />);
    
    // Assert
    expect(app.state('count')).toBe(0);
  });

  it('can increment upwards', () => {
    // Arrange
    let app = shallow(<Counter />);
    let upButton = app.find('a.up');
    expect(upButton.exists()).toBe(true);
    expect(upButton.text()).toBe('+');

    // Act
    upButton.simulate('click');

    // Assert
    expect(app.state('count')).toBe(1);
  });

  it('can decrement downwards', () => {
    // Arrange
    let app = shallow(<Counter />);
    let downButton = app.find('a.down');
    expect(downButton.exists()).toBe(true);
    expect(downButton.text()).toBe('-');

    // Act
    downButton.simulate('click');

    // Assert
    expect(app.state('count')).toBe(-1);
  });

  it('renders consistently', () => {
  const tree = renderer.create(<Counter />).toJSON();
  expect(tree).toMatchSnapshot();
  });
});
