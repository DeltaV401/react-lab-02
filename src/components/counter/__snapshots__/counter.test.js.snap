// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter tests renders consistently 1`] = `
<section
  className="counter"
>
  <span
    className="count "
  >
    0
  </span>
  <br />
  <button
    className="down clicker"
    href="#"
    onClick={[Function]}
  >
    You've been naughty. 
    <br />
    One less present.
  </button>
  <button
    className="up clicker"
    href="#"
    onClick={[Function]}
  >
    You've followed the rules. 
    <br />
    One more present.
  </button>
</section>
`;
